t/03-proxy_wasm/hfuncs/contexts/140-proxy_foreign_function_resolve_lua.t

```diff
@@ -0,0 +1,169 @@
+--- no_error_log
+[crit]
+
+
+
+=== TEST 3: proxy_wasm contexts - call_resolve_lua on_tick
+--- wasm_modules: context_checks
+--- config
+    location /t {
+        proxy_wasm context_checks 'on_tick=call_resolve_lua';
+        return 200;
+    }
+--- error_log
+[error]
+cannot resolve, missing name
+--- no_error_log
+[crit]
+
+
+
+=== TEST 4: proxy_wasm contexts - call_resolve_lua on_http_dispatch_response
+--- wasm_modules: context_checks
+--- config
+    location /t {
+        proxy_wasm context_checks 'on_http_dispatch_response=call_resolve_lua \
+                                   host=127.0.0.1:$TEST_NGINX_SERVER_PORT';
+        return 200;
+    }
+
+    location /dispatch {
+        return 200;
+    }
+--- error_log
+[error]
+cannot resolve, missing name
+--- no_error_log
+[crit]
+
+
+
+=== TEST 5: proxy_wasm contexts - call_resolve_lua on_request_headers
+--- wasm_modules: context_checks
+--- config
+    location /t {
+        proxy_wasm context_checks 'on_request_headers=call_resolve_lua';
+        return 200;
+    }
+--- error_code: 500
+--- ignore_response_body
+--- error_log
+[error]
+cannot resolve, missing name
+--- no_error_log
+[crit]
+
+
+
+=== TEST 6: proxy_wasm contexts - call_resolve_lua on_request_body
+--- load_nginx_modules: ngx_http_echo_module
+--- wasm_modules: context_checks
+--- config
+    location /t {
+        proxy_wasm context_checks 'on_request_body=call_resolve_lua';
+        echo ok;
+    }
+--- request
+POST /t
+payload
+--- error_code: 500
+--- ignore_response_body
+--- error_log
+[error]
+cannot resolve, missing name
+--- no_error_log
+[crit]
+
+
+
+=== TEST 7: proxy_wasm contexts - call_resolve_lua on_response_headers
+--- load_nginx_modules: ngx_http_echo_module
+--- wasm_modules: context_checks
+--- config
+    location /t {
+        proxy_wasm context_checks 'on_response_headers=call_resolve_lua';
+        echo ok;
+    }
+--- ignore_response_body
+--- error_log
+[error]
+can only call resolve_lua during "on_request_headers", "on_request_body", "on_dispatch_response", "on_tick", "on_foreign_function"
+--- no_error_log
+[crit]
+
+
+
+=== TEST 8: proxy_wasm contexts - call_resolve_lua on_response_body
+--- load_nginx_modules: ngx_http_echo_module
+--- wasm_modules: context_checks
+--- config
+    location /t {
+        proxy_wasm context_checks 'on_response_body=call_resolve_lua';
+        echo ok;
+    }
+--- ignore_response_body
+--- error_log
+[error]
+can only call resolve_lua during "on_request_headers", "on_request_body", "on_dispatch_response", "on_tick", "on_foreign_function"
+--- no_error_log
+[crit]
+
+
+
+=== TEST 9: proxy_wasm contexts - call_resolve_lua on_log
+--- wasm_modules: context_checks
+--- config
+    location /t {
+        proxy_wasm context_checks 'on_log=call_resolve_lua';
+        return 200;
+    }
+--- error_log
+[error]
+can only call resolve_lua during "on_request_headers", "on_request_body", "on_dispatch_response", "on_tick", "on_foreign_function"
+--- no_error_log
+[crit]
```

Comment:
Yeah, as I was implementing support for calling `lua_resolver` from `on_http_call_response` I noticed that the code also allowed for calling `dispatch_http_call` from `on_foreign_call` and that `on_foreign_call` would be an additional check in the context tests for the hostcalls.

 There's a test asserting that filter developers can call the `lua_resolver` from the `on_http_call_response` callback; but tests not strictly related to the `lua_resolver` function  (e.g. calling `http_call` from `on_foreign_call` and the new `on_foreign_call` check in the context checks) were left to be addressed in a separate PR as this one is already rather large.
 
 I can include them in this PR. though; if you prefer that way.

I agree that there's value in keeping them separate.

Reply:
>  tests not strictly related to the lua_resolver function (e.g. calling http_call from on_foreign_call and the new on_foreign_call check in the context checks) were left to be addressed in a separate PR as this one is already rather large.

Let's add them in this PR, they belong to this feature and that way it will be done. The PR is not very large to me, it's alright.